Wordle.Api/Wordle.Api/Controllers/LeaderboardController.cs

```diff
@@ -0,0 +1,32 @@
+using Microsoft.AspNetCore.Mvc;
+using Wordle.Api.Models;
+using Wordle.Api.Services;
+
+namespace Wordle.Api.Controllers
+{
+  [Route("api/[controller]")]
+  [ApiController]
+  public class LeaderboardController : ControllerBase 
+  {
+    private readonly WordleService _wordleService;
+
+    public LeaderboardController(WordleService wordleService)
+    {
+      _wordleService = wordleService;
+    }
+
+    [HttpGet("top10")]
+    public IActionResult GetTop10Scores()
+    {
+      var top10Scores = _wordleService.GetTop10Scores();
+      return Ok(top10Scores);
+    }
+
+    [HttpPost]
+    public IActionResult PostScore(Player player)
+    {
+      _wordleService.AddOrUpdatePlayer(player);
+      return Ok();
+    }
```

Comment:
One thing to think about is to make these return a task and also make these async. And for the GET you wanna make it return a task of  Ienumerable/List. One thing to consider for the GET is if you want it to return certain columns or all columns from the table and then just parse what you want into the table and return the top ten results. hint: `.Take(10)`

Reply:
Refactored most of this code and fixed this thank you!